CorinneTest/Factory.cs

```diff
@@ -0,0 +1,195 @@
+                    case 6:
+                        newPerson = MakeChild(lowAge, highAge);
+                        break;
+                    case 7:
+                        newPerson = MakeSeniorWoman(lowAge, highAge);
+                        break;
+                    case 8:
+                        newPerson = MakeSeniorMan(lowAge, highAge);
+                        break;
+                    default:
+                        newPerson = MakeSeniorPerson(lowAge, highAge);
+                        break;
+
+
+                }
+                list.Add(newPerson);
+                i++;
+
+            }
+            return list;
+        }
+        #region various genBirthDates
+        public static DateTime? GenRanAdultBirthDate(int lowAge, int highAge)
+        {
+            if (highAge < 18)
+                return null;
+            var start = new DateTime((2018 - highAge), 5, 29);
+            var minDate = new DateTime((2018 - lowAge), 5, 29);
+            int range = (minDate - start).Days;
+            return start.AddDays(rnd.Next(range));
+        }
+
+        public static DateTime? GenRanChildBirthDate(int lowAge, int highAge)
+        {
+            DateTime maxDate;
+            if (lowAge >= 18) { return null; }
+            if (highAge < 18) { maxDate = new DateTime((2018 - highAge), 5, 29); }
+            else { maxDate = new DateTime(2000, 5, 29); }
+            int range = (DateTime.Today - maxDate).Days;
+            return DateTime.Today.AddDays(rnd.Next(range));
+        }
+
+        public static DateTime? GenRanSeniorBirthDate(int lowAge, int highAge)
+        {
+            DateTime minDate;
+            if (highAge < 6) { return null; }
+            if (lowAge > 65) { minDate = new DateTime((2018 - lowAge), 5, 29); }
+            else { minDate = new DateTime(1953, 5, 29); }
+            var start = new DateTime((2018 - highAge), 5, 29);
+            int range = (minDate - start).Days;
+            return start.AddDays(rnd.Next(range));
+        }
+
+
+        public static DateTime GenRanBirthDate(int lowAge, int highAge)
+        {
+            var start = new DateTime((2018 - highAge), 5, 29);
+            var minDate = new DateTime((2018 - lowAge), 5, 29);
+            int range = (minDate - start).Days;
+            return start.AddDays(rnd.Next(range));
+        }
+        #endregion
+
+        #region various genAges
+        public static int? GenAdultAge(int highAge)
+        {
+            if (highAge < 28)
+                return null;
+            int age = rnd.Next(18, highAge);
+            return age;
+        }
+
+        public static int? GenChildAge(int lowAge)
+        {
+            if (lowAge >= 18)
+                return null;
+            int age = rnd.Next(lowAge, 18);
+            return age;
+        }
+
+        public static int? GenSeniorAge(int highAge)
+        {
+            if (highAge < 65)
+                return null;
+            int age = rnd.Next(65, highAge);
```

Comment:
considering you have GenAge(int low, int high), all of these other methods could return that with unique params

Reply:
i.e. GetSeniorAge could

'return GenAge(65, highAge);'